### ISSUE TITLE
Reject streaming API endpoints if message indicates an error

### ISSUE THREAD

**Issue by clue** [clue/php-docker-react#21]
Currently, we only respect the HTTP status codes to detect error conditions. This works for most error conditions – however, this breaks for streaming API endpoints such as pulling an image.

Pulling a non-existant image returns a success status code, however its status events eventually resolve with a message stream like this:

```
update: {"status":"Pulling repository clue\/does-not-exist"}
update: {"errorDetail":{"message":"Error: image clue\/does-not-exist: not found"},"error":"Error: image clue\/does-not-exist: not found"}
```

Also refs #4

**Comment by clue** [comment]
Is this only applicable to JSON streaming? If so, this should probably be resolved as part of #9.

**Comment by clue** [comment]
Will be closed via #9 (see https://github.com/clue/php-docker-react/commit/4b21ae45f88c7ce20cb7eea0eba12dd9b29c2ea9)

**Comment by clue** [comment]
Closed via #9
